File: Data Structures & Algorithms/depth-of-binary-tree/submission-3.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
void traversal(TreeNode* root, int cnt, int &ans){
    if(!root) return;
    // ans++;
    ans = max(ans, cnt);
    traversal(root->left,cnt+1, ans);
    traversal(root->right,cnt+1, ans);
}
public:
    int maxDepth(TreeNode* root) {
        if(!root) return 0;
        int x =0;
        traversal(root, 1, x);
        return x;
    }
};

File: Data Structures & Algorithms/spiral-matrix-ii/submission-3.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        ans = [[0]*n for _ in range(n)]
        left,right,top, bottom = 0,n-1,0,n-1
        val=1
        while left<=right:
            for c in range(left, right+1):
                ans[top][c]=val
                val+=1
            top+=1
            for c in range (top,bottom+1):
                ans[c][right]=val
                val+=1
            right-=1
            for c in range(right, left-1,-1):
                ans[bottom][c]= val
                val+=1
            bottom-=1
            for c in range(bottom , top-1,-1):
                ans[c][left]=val
                val+=1
            left+=1
        return ans
        